libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/dh-market-participant-edit-actor.component.html

```diff
@@ -67,13 +67,40 @@ <h1 *ngIf="!isEditing">{{ t("createHeader") }}</h1>
     </div>
 
     <!-- Confirmation  -->
-    <div class="button-container" [hidden]="isLoading">
-      <watt-button variant="secondary" (click)="onCancelled()"
-        >{{ t("cancelLabel") }}
-      </watt-button>
-      <watt-button variant="primary" (click)="onSaved()">{{
-        t("saveLabel")
-      }}</watt-button>
-    </div>
+    <ng-container *rxLet="changes$ as changes">
+      <ng-container *rxLet="status$ as status">
+        <div class="button-container" [hidden]="isLoading">
+          <watt-button variant="secondary" (click)="onCancelled()"
+            >{{ t("cancelLabel") }}
+          </watt-button>
+          <watt-button variant="primary" (click)="onSaved(status, changes)">{{
+            t("saveLabel")
+          }}</watt-button>
+        </div>
+        <watt-modal
+          #confirmationModal
+          [size]="'small'"
+          [title]="t('masterData.statusChangeTitle')"
+          [disableClose]="true"
+        >
+          <p>
+            {{
+              t("masterData.statusChangeWarning", {
+                status: t("masterData.statuses." + changes.status),
+                initialStatus: t("masterData.statuses." + status)
+              })
+            }}
+          </p>
+          <watt-modal-actions>
+            <ng-container *transloco="let t">
+              <watt-button (click)="save()">{{ t("yes") }}</watt-button>
+              <watt-button variant="secondary" (click)="closeModal()">{{
+                t("no")
+              }}</watt-button>
```

Comment:
suggestion: another way of doing it which utilizes the `watt-modal` APIs is to use its `close` method. So for the "Yes" button we'd use `(click)="confirmationModal.close(true)"` and for the "No" button we'd use `(click)="confirmationModal.close(false)"`.

Then on the `<watt-modal ..>` itself we'd add `(closed)="onCloseModal($event)"` attribute, which calls the `onCloseModal` in the `dh-market-participant-edit-actor.component.ts` with a param (could be `isAccepted`).

In the component we'd then have:

```ts
onCloseModal(isAccepted: boolean): void {
    if (isAccepted) {
      this.save();
    }
  }
```